geode-core/src/main/resources/META-INF/schemas/geode.apache.org/schema/cache/configurationService.xsd

```diff
@@ -0,0 +1,1301 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one or more
+  ~ contributor license agreements.  See the NOTICE file distributed with
+  ~ this work for additional information regarding copyright ownership.
+  ~ The ASF licenses this file to You under the Apache License, Version 2.0
+  ~ (the "License"); you may not use this file except in compliance with
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~      http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+<xsd:schema
```

Comment:
So we need to check both the cluster-config.xsd, the *.xjb and the generated POJOs into the source tree? It would be ideal if we can do just one. But 
1. keeping the xsd gives us a nice one place to see all the changes we've made to the schema.
2. keeping the POJOs would be easier for the developer to add/modify the methods instead of trying to find the way to transform it using a binding file (.xjb).

Reply:
I'd love to get to the point where we could just generate the POJOs as part of the build process, even if it has to be based on an "internal only" XSD.

During iteration, I found there is some trouble with getting `CacheElement` to selectively apply to some but not all classes.  There are some 3rd party libraries that could help on that front and with gradle integration.  I was hesitant to introduce a new dependency for this, though, and don't remember which licenses are compatible with the Apache 2.0.

I don't like the idea that the config classes are all treated like first-class citizens.  We're already pushing the line where some config classes are special cases.  Every special case is a pitfall for a future developer.

If the massaged XSD isn't actually consumed (and won't be in the future), then I should just remove it from this PR.  Future developers won't remember to update the XSD or bindings file with every update to the POJOs, and once it's out of date, its existence will just add confusion.